src/renderer/dx/DxRenderer.cpp

```diff
@@ -112,6 +283,134 @@ DxEngine::~DxEngine()
     return S_OK;
 }
 
+int DbgPrint(const char* format, ...)
+{
+    static char s_printf_buf[1024];
+    va_list args;
+    va_start(args, format);
+    _vsnprintf_s(s_printf_buf, sizeof(s_printf_buf), format, args);
+    va_end(args);
+    OutputDebugStringA(s_printf_buf);
+    return 0;
+}
+
+Microsoft::WRL::ComPtr<ID3DBlob>
+_CompileShader(
+    std::string source,
+    std::string target,
+    std::string entry = "main")
+{
+    Microsoft::WRL::ComPtr<ID3DBlob> code{};
+    Microsoft::WRL::ComPtr<ID3DBlob> error{};
+
+    HRESULT hr = D3DCompile(
+        source.c_str(),
+        source.size(),
+        nullptr,
+        nullptr,
+        nullptr,
+        entry.c_str(),
+        target.c_str(),
+        0,
+        0,
+        &code,
+        &error);
+
+    if (FAILED(hr))
+    {
+        DbgPrint("D3DCompile failed with %x\n", hr);
+        if (error)
+        {
+            DbgPrint("D3DCompile error: %*s\n", (int)error->GetBufferSize(), (char *)error->GetBufferPointer());
+        }
+
+        THROW_HR(hr);
+    }
+
+    return code;
+}
+
+HRESULT DxEngine::_SetupTerminalEffects()
+{
+    ::Microsoft::WRL::ComPtr<ID3D11Texture2D> swapBuffer;
+    RETURN_IF_FAILED(_dxgiSwapChain->GetBuffer(0, __uuidof(ID3D11Texture2D), (LPVOID*)&swapBuffer));
+
+    // Setup render target.
+    RETURN_IF_FAILED(_d3dDevice->CreateRenderTargetView(swapBuffer.Get(), nullptr, &_renderTargetView));
+
+    // Setup _framebufferCapture, to where we'll copy current frame when rendering effects.
+    D3D11_TEXTURE2D_DESC framebufferCaptureDesc{};
+    swapBuffer->GetDesc(&framebufferCaptureDesc);
+    framebufferCaptureDesc.BindFlags |= D3D11_BIND_SHADER_RESOURCE;
+    RETURN_IF_FAILED(_d3dDevice->CreateTexture2D(&framebufferCaptureDesc, NULL, &_framebufferCapture));
+
+    // Setup the viewport.
+    D3D11_VIEWPORT vp;
+    vp.Width = (FLOAT)_displaySizePixels.cx;

```

Comment:
Pretty sure the static analysis build will get mad at you if you use hard c-style casts like this. Instead we usually prefer 
```suggestion
    vp.Width = static_cast<float>(_displaySizePixels.cx);
```

(or I guess `FLOAT` if there's a real difference between the two for some reason)